### PR TITLE
Fix `gasUsed` and `fee` move balance transaction

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Fixed an edge on how is calculated `gasUsed` and `fee` for some move balance transactions
  
## Proposed changes
- Added an extra check in the component that computes the transaction gas used and fee
- Changed the version of `mx-chain-vm-common-go` module: https://github.com/multiversx/mx-chain-vm-common-go/pull/264

## Testing procedure
- Do a smart contract call with an non hex argument and the transaction `gasUsed` field should be the move balance cost

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/external/transactionAPI/gasUsedAndFeeProcessor.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (gfp *gasUsedAndFeeProcessor) prepareTxWithResultsBasedOnLogs(
 	tx *transaction.ApiTransactionResult,
 	hasRefund bool,
 ) {
-	if tx.Logs == nil {
+	if tx.Logs == nil || (tx.Function == "" && tx.Operation == "transfer") {
```
Comment (iulianpascalau) [inline node/external/transactionAPI/gasUsedAndFeeProcessor.go:55]: do we have a constant for the `transfer` string?

File: outport/process/transactionsfee/transactionsFeeProcessor.go [hunk 2]
```diff
@@ -145,7 +145,10 @@ func (tep *transactionsFeeProcessor) prepareTxWithResultsBasedOnLogs(
 	txWithResults *transactionWithResults,
 	hasRefund bool,
 ) {
-	if check.IfNilReflect(txWithResults.log) {
+	tx := txWithResults.GetTxHandler()
+	res := tep.dataFieldParser.Parse(tx.GetData(), tx.GetSndAddr(), tx.GetRcvAddr(), tep.shardCoordinator.NumberOfShards())
+
+	if check.IfNilReflect(txWithResults.log) || (res.Function == "" && res.Operation == "transfer") {
```
Comment (iulianpascalau) [inline outport/process/transactionsfee/transactionsFeeProcessor.go:151]: also here. `transfer` magic string

File: node/external/transactionAPI/gasUsedAndFeeProcessor.go (hunk 1 above)
Comment (miiu96) [reply node/external/transactionAPI/gasUsedAndFeeProcessor.go]: fixed

File: outport/process/transactionsfee/transactionsFeeProcessor.go (hunk 2 above)
Comment (miiu96) [reply outport/process/transactionsfee/transactionsFeeProcessor.go]: fixed

**Review by iulianpascalau — DISMISSED** [review]

File: outport/process/transactionsfee/transactionsFeeProcessor.go [hunk 3]
```diff
@@ -145,7 +145,10 @@ func (tep *transactionsFeeProcessor) prepareTxWithResultsBasedOnLogs(
 	txWithResults *transactionWithResults,
 	hasRefund bool,
 ) {
-	if check.IfNilReflect(txWithResults.log) {
+	tx := txWithResults.GetTxHandler()
```
Comment (sstanculeanu) [inline outport/process/transactionsfee/transactionsFeeProcessor.go:148]: perhaps add a check for nil tx?

Comment (miiu96) [reply outport/process/transactionsfee/transactionsFeeProcessor.go:149]: added

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**